zapcore/json_encoder.go

```diff
@@ -152,10 +152,14 @@ func (enc *jsonEncoder) AddInt64(key string, val int64) {
 func (enc *jsonEncoder) resetReflectBuf() {
 	if enc.reflectBuf == nil {
 		enc.reflectBuf = bufferpool.Get()
-		enc.reflectEnc = json.NewEncoder(enc.reflectBuf)
-
-		// For consistency with our custom JSON encoder.
-		enc.reflectEnc.SetEscapeHTML(false)
+		// If no EncoderConfig.NewReflectedEncoder is provided by the user, then use default
+		var newReflectedEncoder func(io.Writer) ReflectedEncoder
+		if enc.NewReflectedEncoder == nil {
+			newReflectedEncoder = GetDefaultReflectedEncoder()
+		} else {
+			newReflectedEncoder = enc.NewReflectedEncoder
+		}
```

Comment:
We should do this in `NewJSONEncoder`? We can set `NewReflectedEncoder` in there once if it's nil, and then we don't have to check it on the hot path.